In [1]:
import numpy as np

__Задача1__

Контрольная работа состоит из пяти вопросов. На каждый вопрос приведено четыре варианта ответа, один из которых правильный. Составьте закон распределения числа правильных ответов при простом угадывании. Найдите `M(X)`, `D(X)`.

$$P(X = 0) = C_n^k p^k (1 - p)^{n - k} = C_5^0 \cdot 0.25^0 \cdot 0.75^5 = \frac{5!}{0! \: (5-0)!} \cdot 0.25^0 \cdot 0.75^5 = 0.2373.$$

$$P(X = 1) = C_5^1 \cdot 0.25^1 \cdot 0.75^4 = \frac{5!}{1! \: (5-1)!} \cdot 0.25^1 \cdot 0.75^4 = 0.3955.$$

$$P(X = 2) = C_5^2 \cdot 0.25^2 \cdot 0.75^3 = \frac{5!}{2! \: (5-2)!} \cdot 0.25^2 \cdot 0.75^3 = 0.2637.$$

$$P(X = 3) = C_5^3 \cdot 0.25^3 \cdot 0.75^2 = \frac{5!}{3! \: (5-3)!} \cdot 0.25^3 \cdot 0.75^2 = 0.0879.$$

$$P(X = 4) = C_5^4 \cdot 0.25^4 \cdot 0.75^1 = \frac{5!}{4! \: (5-4)!} \cdot 0.25^4 \cdot 0.75^1 = 0.0146.$$

$$P(X = 5) = C_5^5 \cdot 0.25^5 \cdot 0.75^0 = \frac{5!}{5! \: (5-5)!} \cdot 0.25^5 \cdot 0.75^0 = 0.0010.$$

Закон распределения:

<table border="3">
    <tr>
        <th>$x$</th>
        <td>0</td>
        <td>1</td>
        <td>2</td>
        <td>3</td>
        <td>4</td>
        <td>5</td>
    </tr>
    <tr>
        <th>$P(X = x)$</th>
        <td>0.2373</td>
        <td>0.3955</td>
        <td>0.2637</td>
        <td>0.0879</td>
        <td>0.0146</td>
        <td>0.0010</td>
    </tr>
</table>

Математическое ожидание: M(X) = np



In [5]:
x_values = np.array([0, 1, 2, 3, 4, 5])

print(x_values)

[0 1 2 3 4 5]


In [8]:
x_probabilities = np.array([0.2373, 0.3955, 0.2637, 0.0879, 0.0146, 0.0010])
print(x_probabilities)

[0.2373 0.3955 0.2637 0.0879 0.0146 0.001 ]


In [25]:
m = np.dot(x_values, x_probabilities)
print(m)

1.25


In [27]:
y_values = (x_values - m)**2
print(y_values)

[ 1.5625  0.0625  0.5625  3.0625  7.5625 14.0625]


In [28]:
d = np.dot(y_values, x_probabilities)
print(d)


0.9375


__Ответ:__ 
$$Закон  распределения:$$

<table border="3">
    <tr>
        <th>$x$</th>
        <td>0</td>
        <td>1</td>
        <td>2</td>
        <td>3</td>
        <td>4</td>
        <td>5</td>
    </tr>
    <tr>
        <th>$P(X = x)$</th>
        <td>0.2373</td>
        <td>0.3955</td>
        <td>0.2637</td>
        <td>0.0879</td>
        <td>0.0146</td>
        <td>0.0010</td>
    </tr>
</table>

$$Математическое ожидание: М(X) = 1,25$$
$$Дисперсия: D(X) = 0.9375$$

__Задача2__

Вероятность поймать рыбу при одном забрасывании удочки равна 0.01. Какова вероятность того, что число пойманных рыб при 200 забросах удочки будет не менее 5 и не более 10?

In [30]:
p = 0.01
n = 200

In [32]:
lambda_ = p * n
lambda_

2.0

In [2]:
def poisson_proba(k, lambda_):
    return (lambda_ ** k) * (np.exp(-lambda_)) / np.math.factorial(k)

Вероятность поимки от 5 до 10 рыб равна сумме вероятностей поимки каждого количества рыб

In [35]:
answer = sum(poisson_proba(k=i, lambda_=lambda_) for i in range(5, 11))
answer

0.052644709119342686

__Ответ:__ 0.053

__Задача 3__

В условиях предыдущей задачи сколько забросов удочки нужно сделать чтобы быть уверенным с вероятностью 0.9, что хотя бы 10 раз будет поймана рыба?

In [37]:
p = 0.01
n = 1
probabilities = 0

In [41]:
while probabilities <= 0.9:
    lambda_ = p * n
    probabilities = 1 - sum(poisson_proba(k=i, lambda_=lambda_) for i in range(11))
    n += 1
print(countn, probabilities)


1542 0.900141980664332


__Ответ:__ 1542